docs/api/proposed_endpoints.md

```diff
@@ -1,5 +1,152 @@
+last_name   | text      | last name of the user
+email_id    | text      | email of user. Errors out if email already registered.
+password    | text      | Hash of the user's password
+gender      | choices   | gender (Like Male, Female, Others)
+tshirt_size | choices   | size of the TShirt (Like Small, Medium etc.)
+contact     | text      | contact number
+ticket_id   | text      | ticket ID generated at registration
+created_at  | datetime  | date and time of registration
+modified_at | datetime  | date and time of modification
+user_type   | choices   | Type of user like admin, volunteer, attendee
+
+
+
+**Request**
+```json
+
+{
+    "id": null,
+    "first_name": "John",
+    "last_name": "Hawley",
+    "email_id": "john@localhost.com",
+    "password": "VerySafePassword0909",
+    "gender": "Male",
+    "tshirt_size": "Medium",
+    "contact": "9999999999",
+    "ticket_id": null,
+    "created_at": null,
+    "modified_at": null,
+    "user_type": null
+}
+```
+
+
+**Response**
+```json
+
+Status: 201 Created
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "first_name": "John",
+    "last_name": "Hawley",
+    "email_id": "john@localhost.com",
+    "auth_token": "eyJ0eXAi0iJKV1QiLCJh",
+    "gender": "Male",
+    "tshirt_size": "Medium",
+    "contact": "9999999999",
+    "ticket_id": "This-is-a-unique-ticket-id",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-01T17:30:42Z",
+    "user_type": "Attendee"
+}
+```
+
+# Current user actions
+
+## Get user's details
+
+```
+GET /api/me (requires authentication)
+```
+
+**Response**
+```json
+
+Status: 200 OK
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "first_name": "Jaun",
+    "last_name": "Hawley",
+    "email_id": "jaun@localhost.com",
+    "gender": "Male",
+    "tshirt_size": "Extra Large",
+    "contact": "9999999998",
+    "ticket_id": "This-is-a-unique-ticket-id",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-02T07:10:42Z",
+    "user_type": "Volunteer"
+}
+```
+
+## Update user details
+
+```
+PATCH /api/me (requires authentication)
+```
+
+**Request**
+```json
+
+{
+    "first_name": "Jaun",
+    "last_name": "Hawley",
+    "email_id": "jaun@localhost.com",
+    "gender": "Male",
+    "tshirt_size": "Extra Large",
+    "contact": "9999999998",
+    "user_type": "Volunteer"
+}
+```
+
+**Response**
+```json
+
+Status: 200 OK
+{
+    "id": "0f342ac1-ac32-4bd1-3612-efa32bc3d9a0",
+    "first_name": "Jaun",
+    "last_name": "Hawley",
+    "email_id": "jaun@localhost.com",
+    "gender": "Male",
+    "tshirt_size": "Extra Large",
+    "contact": "9999999998",
+    "ticket_id": "This-is-a-unique-ticket-id",
+    "created_at": "2018-08-01T17:30:42Z",
+    "modified_at": "2018-08-02T07:10:42Z",
+    "user_type": "Volunteer"
+}
+```
+
+## Delete User
+
+```
+DELETE /api/me (requires authentication)
+```
+
+_NOTE_: This will be a soft-delete.
+
+**Response**
+```json
+
+Status: 204 No-Content
```

Comment:
You can just say `Status: 204 No-Content`

No need to wrap this in json.

Also, you can do this outside of json formatter. Ideally it should only have JSON response.